kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/k8s/HelmClient.java

```diff
@@ -0,0 +1,158 @@
+     * @return the helm client
+     */
+    public HelmClient install(String chartName, String releaseName, String version, Map<String, String> valuesMap) {
+        LOGGER.info("Installing helm-chart {}", releaseName);
+        String values = valuesMap.entrySet().stream()
+                .map(entry -> String.format("%s=%s", entry.getKey(), entry.getValue()))
+                .collect(Collectors.joining(","));
+        this.version = version;
+        Exec.exec(null, wait(namespace(version(command("install",
+                releaseName,
+                "--set", values,
+                "--timeout", INSTALL_TIMEOUT_SECONDS,
+                "--debug",
+                chartName)))), 0, true);
+        return this;
+    }
+
+    /**
+     * Add repository to the helm client.
+     *
+     * @param repoName the repo name
+     * @param repoUrl the repo url
+     * @return the helm client
+     */
+    public HelmClient addRepository(String repoName, String repoUrl) {
+        LOGGER.info("Adding repo {}", repoName);
+        Exec.exec(null, command("repo", "add",
+                repoName,
+                repoUrl), 0, true);
+        return this;
+    }
+
+    /** Delete a chart given its release name from the currently configured namespace
+     * @param releaseName the release name
+     * @return the helm client
+     */
+    public HelmClient delete(String releaseName) {
+        LOGGER.info("Deleting helm-chart {}", releaseName);
+        delete(namespace, releaseName);
+        return this;
+    }
+
+    /**
+     * Delete a Helm chart in specific namespace by given release name
+     * @param namespace namespace where chart is installed
+     * @param releaseName helm chart release name
+     * @return this helm client
+     */
+    public HelmClient delete(String namespace, String releaseName) {
+        LOGGER.info("Deleting helm-chart:{} in namespace:{}", releaseName, namespace);
+        Exec.exec(null, wait(command("uninstall", releaseName, "--namespace", namespace)), 0, true);
+        return this;
+    }
+
+    private List<String> command(String... rest) {
+        List<String> result = new ArrayList<>();
+        result.add(helmCommand);
+        result.addAll(asList(rest));
+        return result;
+    }
+
+    private List<String> version(List<String> args) {
+        if (version != null && !version.isEmpty() && !version.equalsIgnoreCase("latest")) {
+            args.add("--version");
+            args.add(version);
+        }
+        return args;
+    }
+
+    /** Sets namespace for client */
+    private List<String> namespace(List<String> args) {
+        if (namespace == null) {
+            LOGGER.warn("Namespace has not been set! Helm client will run the command on the default namespace");
+            return args;
+        }
+        args.add("--namespace");
+        args.add(namespace);
+        return args;
+    }
+
+    private List<String> wait(List<String> args) {
+        args.add("--wait");
```

Comment:
in general, assuming a caller is passing you a mutable is a bad thing.  a safer way is to create a copy, mutate.